server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -140,34 +152,12 @@ public ComposableIndexTemplate(
         @Nullable DataStreamTemplate dataStreamTemplate,
         @Nullable Boolean allowAutoCreate
     ) {
-        this(indexPatterns, template, componentTemplates, priority, version, metadata, dataStreamTemplate, allowAutoCreate, null);
-    }
-
-    ComposableIndexTemplate(
-        List<String> indexPatterns,
-        @Nullable Template template,
-        @Nullable List<String> componentTemplates,
-        @Nullable Long priority,
-        @Nullable Long version,
-        @Nullable Map<String, Object> metadata,
-        @Nullable DataStreamTemplate dataStreamTemplate,
-        @Nullable Boolean allowAutoCreate,
-        @Nullable List<String> ignoreMissingComponentTemplates
-    ) {
-        this(
-            indexPatterns,
-            template,
-            componentTemplates,
-            priority,
-            version,
-            metadata,
-            dataStreamTemplate,
-            allowAutoCreate,
-            ignoreMissingComponentTemplates,
-            null
-        );
+        this(indexPatterns, template, componentTemplates, priority, version, metadata, dataStreamTemplate, allowAutoCreate, null, null);
     }
 
+    /**
+     * @deprecated use {@link Builder} instead
+     */
     public ComposableIndexTemplate(
```

Comment:
I think this one wants a `@Deprecated(forRemoval = true)` too, presumably we're going to make it `private` in a follow-up.